pkg/credentialprovider/shim/plugin_executor.go

```diff
@@ -0,0 +1,304 @@
+// Copyright 2022 D2iQ, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package shim
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"golang.org/x/sync/singleflight"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"k8s.io/klog/v2"
+	"k8s.io/kubelet/config/v1beta1"
+	credentialproviderapi "k8s.io/kubelet/pkg/apis/credentialprovider"
+
+	"github.com/mesosphere/kubelet-image-credential-provider-shim/apis/config/v1alpha1"
+	"github.com/mesosphere/kubelet-image-credential-provider-shim/pkg/log"
+	"github.com/mesosphere/kubelet-image-credential-provider-shim/pkg/urlglobber"
+)
+
+// newPluginProvider returns a new pluginProvider based on the credential provider config.
+func newPluginProvider(
```

Comment:
I was trying to update the demo script but I don't think I understand how this works.

I see the plugins were copied out to the host, but is the shim provider supposed to be copied out too?
```
root@image-credential-provider-test-control-plane:/etc/kubernetes# ll image-credential-provider
total 50244
drwxr-xr-x 7 root root      224 Nov 18 21:26 ./
drwxr-xr-x 1 root root     4096 Nov 18 21:10 ../
-rwxr-xr-x 1 root root 15015936 Nov 18 21:26 acr-credential-provider*
-rwxr-xr-x 1 root root 15351808 Nov 18 21:26 ecr-credential-provider*
-rwxr-xr-x 1 root root 10391552 Nov 18 21:26 gcr-credential-provider*
-rwxr-xr-x 1 root root 10678272 Nov 18 21:26 static-credential-provider*
```

Reply:
I haven't tried it via demo script... Forgot all about it! Good idea to update it!  I'll see what I can do with this as part of this pr.

Reply:
Updated demo script!